VTUL/dlp-access
# CSV Export

**JIRA Ticket**: (link) (:star:)
https://webapps.es.vt.edu/jira/browse/LIBTD-2628

# What does this Pull Request do? (:star:)
Allows metadata for all items, all collections, and items from top-level collections to be downloaded in a CSV file.
also includes a fix to the SiteForm component and adds a default value for viewOption.

# What's the changes? (:star:)
-test for the component
-adds a package to handle creating the CSV file
-src/lib/available_attributes.js - the fields to be included in the CSV spreadsheet
-src/pages/admin/CSVExport.js - the csv export component
-src/pages/admin/SiteAdmin.js - adds the component to the side menu
-src/pages/admin/SiteForm.js - unrelated fix for default value for viewOption.

# How should this be tested?
-Click on CSV export in the side menu, the page should list the top level collections and options for all collection and all items upon loading. If all items is checked, no other collections should be checked, it all collections is checked no options under the items section should be checked (collection and item records have different metadata fields and shouldn't be combined into a singe CSV). Click "Generate CSV" button, and then link should appear below. Clicking the link should cause the file to download and the link to disappear. You can now repeat, choose a different report, generate it, and download the new report should download.

# Additional Notes:
branch is LIBTD-2628

# Interested parties
@whunter 

(:star:) Required fields
